axiom/optimizer/QueryGraph.h

```diff
@@ -889,13 +889,24 @@ struct BaseTable : public PlanObject {
 using BaseTableCP = const BaseTable*;
 
 struct ValuesTable : public PlanObject {
-  explicit ValuesTable(const logical_plan::ValuesNode& values)
-      : PlanObject{PlanType::kValuesTableNode}, values{values} {}
+  using Variants = const std::vector<velox::Variant>*;
+  using Vectors = const std::vector<velox::RowVectorPtr>*;
+  using Data = std::variant<Variants, Vectors>;
+
+  explicit ValuesTable(const velox::Type* _dataType, Data _data)
+      : PlanObject{PlanType::kValuesTableNode},
+        dataType{_dataType},
+        data{std::move(_data)},
+        cardinality_{cardinality(data)} {
+    VELOX_CHECK_NOT_NULL(dataType);
+  }
 
   /// Correlation name, distinguishes between uses of the same values node.
   Name cname{nullptr};
 
-  const logical_plan::ValuesNode& values;
+  const velox::Type* dataType;
```

Comment:
```suggestion
  const velox::Type& dataType;
```
it cannot be null, right?
Why